src/powfacpy/plot_interface.py

```diff
@@ -90,19 +94,44 @@ def plot_monitored_variables(self,obj,variables,
     for var in variables:
       data_series.AddCurve(obj,var)
       self.set_curve_attributes(data_series,**kwargs)
+    x_axis = self.get_x_axis_of_active_plot()
+    self.set_x_axis_attributes(x_axis, **kwargs)
     self.active_graphics_page.Show()
 
-  def get_data_series_of_active_plot(self):
-    """Get the dataseries of the currently active plot.
+
+  def _get_element_of_active_plot(self, element):
+    """Get sub-element of the currently active plot.
     """
     try:
-      return self.active_plot.GetDataSeries()
+      if element == 'data_series':
+        return self.active_plot.GetDataSeries()
+      elif element == 'x_axis':
+        return self.active_plot.GetAxisX()
+      elif element == 'y_axis':
+        return self.active_plot.GetAxisY()
+
```

Comment:
my suggestion is to give the specific function (e. g. self.active_plot.GetDataSeries) to an error handling function (e. g. _handle_possible_no_plot_activated_error()), which calls it and return its output -- or handles the error, otherwise. 

like this: 

 ```suggestion
   def _handle_possible_no_plot_activated_error(self, get_from_plot_function):
    """Handle error if no active plot available.
    """
    try:
      return get_from_plot_function()
    except(AttributeError) as e:
      if not self.active_plot:
        raise powfacpy.PFNoPlotActivatedError()
      else:
        raise AttributeError(e)
 
def get_data_series_of_active_plot(self):
    """Get the dataseries of the currently active plot.
    """
    return _handle_possible_no_plot_activated_error(
      self.active_plot.GetDataSeries)
```